pkg/steward/steward.go

```diff
@@ -55,37 +52,31 @@ func New(getter storage.Getter, t traversal.Traverser, r retrieval.Interface, p
 // It assumes all chunks are available locally. It is therefore
 // advisable to pin the content locally before trying to reupload it.
 func (s *steward) Reupload(ctx context.Context, root swarm.Address) error {
-	sem := make(chan struct{}, parallelPush)
-	eg, _ := errgroup.WithContext(ctx)
+	uploaderSession := s.netStore.DirectUpload()
+	getter := s.netStore.Download(false)
+
 	fn := func(addr swarm.Address) error {
-		c, err := s.getter.Get(ctx, storage.ModeGetSync, addr)
+		c, err := getter.Get(ctx, addr)
+		if err != nil {
+			return err
+		}
+
+		err = uploaderSession.Put(ctx, c)
```

Comment:
No need to check the error, the following will do: `return uploaderSession.Put(ctx, c)`.